### PR TITLE
changed contact us chat trigger, added downgrade chat trigger

### PR DESCRIPTION
**Pull request by EugeneTseitlin**
Close: #4836 

## PR Checklist

- [ ] Tests for the changes have been added
- [ ] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

File: packages/amplication-client/src/Purchase/PurchasePage.tsx [hunk 1]
```diff
@@ -78,6 +79,11 @@ const PurchasePage = (props) => {
     });
   }, [openHubSpotChat, currentWorkspace.id]);
 
+  const handleDowngradeClick = useCallback(() => {
+    history.push("?downgrade=true");
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Purchase/PurchasePage.tsx:83]: @EugeneTseitlin, I assume the URL is required for Hubspot to identity the specific chat flow to open? if that is correct, please make sure it is documented here in the code otherwise, someone in the future can easily remove it without understanding the need.

File: packages/amplication-client/src/Purchase/PurchasePage.tsx [hunk 2]
```diff
@@ -70,6 +70,7 @@ const PurchasePage = (props) => {
     });
 
   const handleContactUsClick = useCallback(() => {
+    history.push("?contact-us=true");
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Purchase/PurchasePage.tsx:73]: Not part of this PR - but I think the event's name is not good enough - "contact us" is generic and already used in the top bar. 
Please change it to "PurchaseContactUsButtonClicked" (new event - do not change the name of the current event if it is used in other places)

File: packages/amplication-client/src/Purchase/PurchasePage.tsx (hunk 1 above)
Comment (yuval-hazaz) [reply packages/amplication-client/src/Purchase/PurchasePage.tsx:83]: also, please add event "PurchaseDowngradeClick"

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]

**Comment by yuval-hazaz** [comment]
@GreenMachine01 please have a look at the comments here about the events, and see if it is aligned with your plan

**Comment by GreenMachine01** [comment]
@yuval-hazaz I think we went with the generic CTA click event that has a param for the action
image

**Comment by yuval-hazaz** [comment]
> @yuval-hazaz I think we went with the generic CTA click event that has a param for the action image

Isn't that the case when we decided to go with separate events because we wanted to understand each one separately?

File: packages/amplication-client/src/Purchase/PurchasePage.tsx (hunk 1 above)
Comment (EugeneTseitlin) [reply packages/amplication-client/src/Purchase/PurchasePage.tsx:83]: > also, please add event "PurchaseDowngradeClick"

This event has already exist.

**Comment by GreenMachine01** [comment]
@yuval-hazaz we agreed after this event was already out. we didn't go back to change it.
we can open a new ticket with the requested changes.

File: packages/amplication-client/src/Purchase/PurchasePage.tsx [hunk 3]
```diff
@@ -117,6 +123,9 @@ const PurchasePage = (props) => {
           setLoading(true);
           await upgradeToPro(selectedBillingPeriod, intentionType);
           break;
+        case "plan-amplication-free":
```
Comment (barshimi) [inline packages/amplication-client/src/Purchase/PurchasePage.tsx:126]: why we're keep using switch/case it's a bad habit.

File: packages/amplication-client/src/Purchase/PurchasePage.tsx (hunk 1 above)
Comment (yuval-hazaz) [reply packages/amplication-client/src/Purchase/PurchasePage.tsx:83]: @EugeneTseitlin 
comments in code

Comment (EugeneTseitlin) [reply packages/amplication-client/src/Purchase/PurchasePage.tsx:83]: comments are added, event that you meant is "PricingPageCTAClick" with "intentionType": "dowgrade" is issed on line 111.

**Review by yuval-hazaz — APPROVED** [review]
